nearcore/src/state_sync.rs

```diff
@@ -662,12 +668,21 @@ fn get_latest_epoch(
     let hash = head.last_block_hash;
     let header = chain.get_block_header(&hash)?;
     let final_hash = header.last_final_block();
-    let sync_hash = StateSync::get_epoch_start_sync_hash(chain, final_hash)?;
     let final_block_header = chain.get_block_header(&final_hash)?;
     let epoch_id = *final_block_header.epoch_id();
     let epoch_info = epoch_manager.get_epoch_info(&epoch_id)?;
     let prev_epoch_id = epoch_manager.get_prev_epoch_id_from_prev_block(&head.prev_block_hash)?;
     let epoch_height = epoch_info.epoch_height();
+
+    // This is not the way protocol features are normally gated, but this is not actually
+    // a protocol feature/change. The protocol version is just an easy way to coordinate
+    // among nodes for now
+    let sync_hash =
+        if checked_feature!("stable", StateSyncHashUpdate, epoch_info.protocol_version()) {
+            chain.get_current_epoch_sync_hash(&final_block_header)?
+        } else {
+            Some(chain.get_epoch_start_sync_hash(final_hash)?)
+        };
```

Comment:
I like this pattern a lot. I would expect it to appear in all the places where we need to figure out the right sync hash. Would it make sense to refactor this to a method? Ideally the feature check should happen in just one place and then the logic should depend on the type of state sync that is encoded in the structs. 

Also the naming is a bit inconsistent - how about the following? Would be nice to make those private too. The general idea is to pick a name for each type of state sync and use those names everywhere. I think you're doing it already with `current epoch state sync` which is nice. 
* get_current_epoch_sync_hash 
* get_previous_epoch_sync_hash

If it so happens that the method is reused for getting the sync hash and the first block of an epoch - let's just add a new method for sync hash and call the one for getting epoch's first block.

Reply:
yea good idea